### PR TITLE
MC loading bar

### PR DESCRIPTION
**Pull request by Cyberboss**
:cl: Cyberboss
add: Subsystem initializations now come with a progress bar
/:cl:

TODO:
- [ ] Actually get a graphic for it

### CODE REVIEW

**Comment by Jordie0608** [comment]
Well cool, but why?

**Comment by Cyberboss** [comment]
Because cool

**Comment by MrStonedOne** [comment]
just use the existing progress bar system, only transform it.

File: code/_onclick/hud/screen_objects.dm [hunk 1]
```diff
@@ -701,4 +701,99 @@
...
+	var/F = file("data/subsystem_average_init_times.json")
+	if(F)
+		F = file2text(F)
+		if(F)
+			average_subsystem_init_time = json_decode(F)
+	LAZYINITLIST(average_subsystem_init_time)
+	var/list/asit = average_subsystem_init_time
+
+	if(asit.len != number_of_ss)
+		asit.Cut()		//rebuild the cache
```
Comment (MrStonedOne) [inline code/_onclick/hud/screen_objects.dm]: na, let it just assume 5 seconds.

**Comment by Cyberboss** [comment]
I want to use a larger, smoother progress bar for this. The regular one only has 20 positions

**Comment by MrStonedOne** [comment]
cyberboss, fix our current one then, have it do actual offset pixels as overlays, and maybe even have it do some alpha work on the last most pixel to give it more states.

**Comment by F-OS** [comment]
screenshots?

**Comment by Cyberboss** [comment]
Ill try and finish it tonight

**Comment by Jordie0608** [comment]
That's not what finish means.

**Pull request closed without merging**